src/shims/env.rs

```diff
@@ -27,24 +28,24 @@ impl EnvVars {
             for (name, value) in env::vars() {
                 if !excluded_env_vars.contains(&name) {
                     let var_ptr =
-                        alloc_env_var(name.as_bytes(), value.as_bytes(), &mut ecx.memory);
-                    ecx.machine.env_vars.map.insert(name.into_bytes(), var_ptr);
+                        alloc_env_var_as_c_str(name.as_ref(), value.as_ref(), ecx);
```

Comment:
```suggestion
                        alloc_env_var_as_c_str(&name, &value, ecx);
```
I think this should also work.

Reply:
> 
> 
> I think this should also work.

I tried the suggested change in my local build, and got the attached error below.
![image](https://user-images.githubusercontent.com/10286488/71417932-145dc300-2636-11ea-86c8-1f9ecb1c32a0.png)

The `.as_ref()` call is from `impl AsRef<OsStr> for String`. Is there a way to make this more explicit in code?

Reply:
Ah, good point, I hadn't seen that.

One option would be `OsStr::new`. But in this case, `.as_ref()` also seems fine then, so let's just stick to that.